### ISSUE TITLE
TypeError: Qt designer dumps core while editing PyDMEmbeddedDisplay

### ISSUE THREAD

**Issue by prjemian** [slaclab/pydm#565]
**Describe the bug**

Qt designer crashes when starting to edit the `filename` property of a `PyDMEmbeddedDisplay` widget.

**Expected behavior**

Working on [*adl2pydm*](https://github.com/BCDA-APS/adl2pydm) translator and need to translate MEDM's *embedded display* widget.  Also, the *composite* widget can be used like this, as well.

**Steps to Reproduce**

![Clipboard01](https://user-images.githubusercontent.com/2279984/67922474-d2a22600-fb78-11e9-90a1-4775b38ab396.jpg)

1. start `designer`
1. create a new dialog screen
1. place a `PyDMEmbeddedDisplay` widget on the display
1. start to type into the `filename` property
1. core dump now, here is my console report:

```
(bluesky-pydm) mintadmin@mint-vm:~/.../eclipse/adl2pydm$ designer &
[10] 17126
(bluesky-pydm) mintadmin@mint-vm:~/.../eclipse/adl2pydm$ Traceback (most recent call last):
  File "/home/mintadmin/Apps/anaconda/envs/bluesky-pydm/lib/python3.7/site-packages/pydm/widgets/embedded_display.py", line 124, in filename
    self.load_if_needed()
  File "/home/mintadmin/Apps/anaconda/envs/bluesky-pydm/lib/python3.7/site-packages/pydm/widgets/embedded_display.py", line 140, in load_if_needed
    self.embedded_widget = self.open_file()
  File "/home/mintadmin/Apps/anaconda/envs/bluesky-pydm/lib/python3.7/site-packages/pydm/widgets/embedded_display.py", line 161, in open_file
    (filename, extension) = os.path.splitext(full_fname)
  File "/home/mintadmin/Apps/anaconda/envs/bluesky-pydm/lib/python3.7/posixpath.py", line 122, in splitext
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType

[10]+  Aborted                 (core dumped) designer
```

**Possible Solution**
?

**My Platform**

```
(bluesky-pydm) mintadmin@mint-vm:~/.../eclipse/adl2pydm$ more /etc/os-release 
NAME="Linux Mint"
VERSION="19.2 (Tina)"
ID=linuxmint
ID_LIKE=ubuntu
PRETTY_NAME="Linux Mint 19.2"
VERSION_ID="19.2"
HOME_URL="https://www.linuxmint.com/"
SUPPORT_URL="https://forums.ubuntu.com/"
BUG_REPORT_URL="http://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/"
PRIVACY_POLICY_URL="https://www.linuxmint.com/"
VERSION_CODENAME=tina
UBUNTU_CODENAME=bionic
```

```
(bluesky-pydm) mintadmin@mint-vm:~/.../eclipse/adl2pydm$ conda list python
# packages in environment at /home/mintadmin/Apps/anaconda/envs/bluesky-pydm:
#
# Name                    Version                   Build  Channel
ipython                   7.9.0            py37h5ca1d4c_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
python                    3.7.3                h33d41f4_1    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
```

```
(bluesky-pydm) mintadmin@mint-vm:~/.../eclipse/adl2pydm$ conda list pydm
# packages in environment at /home/mintadmin/Apps/anaconda/envs/bluesky-pydm:
#
# Name                    Version                   Build  Channel
pydm                      1.7.3                      py_0    conda-forge
```

**Additional context**

Reproduce my conda environment following my [instructions](https://github.com/BCDA-APS/adl2pydm/tree/master/pyqtdesignerpath#install-pydm--bluesky-in-custom-environment).

Looked at #481 and decided that issue should be closed.  It confirmed `PyDMEmbeddedDisplay` is the right widget to use.

The translator is showing some solid progress but more work is needed.  Here's the motorx screen from synApps:
![motorx](https://user-images.githubusercontent.com/2279984/67922450-c7e79100-fb78-11e9-9297-c031319dfd05.jpg)

**Comment by prjemian** [comment]
Actually, MEDM [does not have an *embedded display* widget](https://github.com/BCDA-APS/adl2pydm/issues/17).  It has a *composite* widget which has at least two different characters:

1. container for a collection of widgets
1. embed another display by name and macros (which has a collection of widgets)

These map, respectively, to these PyDM widgets:

1. PyDMFrame
1. PyDMEmbeddedDisplay

**Comment by prjemian** [comment]
Here is an example (from areaDetector ADSimDetector) showing the use of MEDM's *composite* widget to embed the content of several files from ADCore (and purposely, the ADShutter.ui screen was not converted to show how that is handled):
![Clipboard01](https://user-images.githubusercontent.com/2279984/67965737-9bf7fa00-fbd0-11e9-9f9a-dd6bd55f8767.jpg)


(Note the *adl2pydm* converter is still a work in progress.)

**Comment by prjemian** [comment]
When *designer* opens the `.ui` file above, it crashes immediately with the same error.

**Comment by klauer** [comment]
I think pydm should configure `sys.excepthook` + `threading.excepthook` when it's loaded in the designer to avoid similar crashes. 

The error could be propagated to the user using something along the lines of https://github.com/NSLS-II/mily/blob/3ec67fc64a58df3478d01cfbe692639eb682cd10/mily/utils.py#L56

**Comment by hhslepicka** [comment]
@prjemian this converter is looking great! Thank you for all the effort putting that together.
I will address this bug.
@klauer thanks for the suggestion. I will take that into consideration as well.

**Comment by hhslepicka** [comment]
@prjemian I think that this bug was addressed with https://github.com/slaclab/pydm/commit/495cb986a27852e7ad4127ba7212093c6f3584f4
Do you mind to give it a try using the code from master?
We plan to cut a bugfix release soon.

**Comment by prjemian** [comment]
Now working from PyDM master branch pulled within the hour (1.7.3+70.g35aeb22f), confirm this reported bug does not appear.  I'll close this issue as having been resolved previously (495cb98 via PR #564).

Thanks!
